openpype/lib/delivery.py

```diff
@@ -4,27 +4,41 @@
 import glob
 import clique
 import collections
+import re
 
 
 def collect_frames(files):
     """
         Returns dict of source path and its frame, if from sequence
 
-        Uses clique as most precise solution
+        Uses clique as most precise solution, used when anatomy template that
+        created files is not known.
+
+        Depends that version substring starts with 'v' with any number of
+        numeric characters after.
 
         Args:
-            files(list): list of source paths
+            files(list) or (set with single value): list of source paths
         Returns:
             (dict): {'/asset/subset_v001.0001.png': '0001', ....}
     """
     collections, remainder = clique.assemble(files, minimum_items=1)
 
+    real_file_name = None
     sources_and_frames = {}
+    if len(files) == 1:
+        real_file_name = list(files)[0]
+        sources_and_frames[real_file_name] = None
```

Comment:
A single file could still be a frame range based render - we'd still like to collect the actual frame for that file. No?